File: ProductManagementSystem.Domain/Entities/Product.cs
using ProductManagementSystem.Domain.Common;

namespace ProductManagementSystem.Domain.Entities;

public enum EmployeeDepartment
{
    Development,
    Design,
    Testing,
    HR
}

public class Product : BaseEntity
{
    public string Name { get; set; }
    public decimal Price { get; set;}
    public string ISBN { get; set; }
    public string UserId { get; set; }
    public ApplicationUser CreatedBy { get; set; }
    public ApplicationUser UpdatedBy { get; set; }
    public string CategoryId { get; set; }
    public Category Category { get; set; }

}

File: productmanagementsystem.client/src/models/entities/product.ts
import { ICategoryDTO } from './category';

export interface IProductDTO {
  id: string;
  name: string;
  price: number;
  iSBN: string;
  userId: string;
  categoryId: string;
  category: ICategoryDTO;
}

export interface CreateOrUpdateProductFormData {
  name: string;
  price: string;
  ISBN: string;
  categoryId: string;
}

export interface CreateProductRequest {
  name: string;
  price: number;
  iSBN: string;
  categoryId: string;
}

export interface UpdateProductRequest {
  id: string;
  name: string;
  price: number;
  iSBN: string;
  categoryId: string;
}
